### ISSUE TITLE
MathJax configuration should appear before loading MathJax itself

### ISSUE THREAD

**Issue by juliangilbey** [sphinx-doc/sphinx#9878]
### Describe the bug

According to the MathJax docs (see v2.7: http://docs.mathjax.org/en/v2.7-latest/configuration.html#using-in-line-configuration-options and v3.2: http://docs.mathjax.org/en/v3.2-latest/web/configuration.html#configuration-using-an-in-line-script), it is necessary to run the configuration **before** loading MathJax itself.  In https://github.com/sphinx-doc/sphinx/blob/647314133d7a9a524087eddd9c21203010cb4aa7/sphinx/ext/mathjax.py#L82-L98 though, the `app.add_js_file()` method for the main MathJax.js script is called before the config scripts are added; the solution to this would be to move the first part of the code here, dealing with the MathJax.js script, after the two `if` blocks which handle the `mathjax2_config` and `mathjax3_config`.

### How to Reproduce

```
$ git clone https://github.com/juliangilbey/abydos.git
$ cd abydos
$ git checkout sphinxbug
$ pip install cython
$ pip install -r requirements-dev.txt
$ pip install abydos
$ cd docs
$ make html SPHINXOPTS="-D language=de"
$ # open _build/html/abydos.distance.html and see the MathJax header lines, somewhere around line 18
```
(Note that there may be a lot of "undefined label" warnings, but I do not yet understand the source of these; this is an unrelated issue.)

### Expected behavior

The MathJax lines in the above html file appear as:
```html
        <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@2.7.9/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
        <script type="text/x-mathjax-config">MathJax.Hub.Config("{tex2jax: {inlineMath: [ ['$','$'], ['\\(','\\)'] ], processEscapes: true}}")</script>
```
but should appear in the opposite order as:
```html
        <script type="text/x-mathjax-config">MathJax.Hub.Config("{tex2jax: {inlineMath: [ ['$','$'], ['\\(','\\)'] ], processEscapes: true}}")</script>
        <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@2.7.9/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
```

### Your project

https://github.com/juliangilbey/abydos.git

### OS

Linux

### Python version

3.9

### Sphinx version

4.2.0 (same issue occurs with 4.3.0)

### Sphinx extensions

sphinx.ext.mathjax

**Comment by juliangilbey** [comment]
Oh, and it turns out this fix, while harmless, is not actually needed, because the `defer` option means that `MathJax.js` is not actually loaded until later. But this order is the way it appears in the MathJax documentation, so it might be beneficial anyway. Sorry for the unnecessary issue, and thanks for dealing with it so efficiently.
